checker/src/main/java/org/checkerframework/checker/nullness/NullnessTransfer.java

```diff
@@ -371,7 +369,8 @@ public TransferResult<NullnessValue, NullnessStore> visitMethodInvocation(
         List<AnnotatedTypeMirror> methodParams = methodType.getParameterTypes();
         List<? extends ExpressionTree> methodArgs = tree.getArguments();
         for (int i = 0; i < methodParams.size() && i < methodArgs.size(); ++i) {
-            if (methodParams.get(i).hasAnnotation(NONNULL)) {
+            if (methodParams.get(i).hasAnnotation(NONNULL)
+                    && !methodArgs.get(i).toString().equals(receiver.getTree().toString())) {
```

Comment:
Definitely do not add such string comparisons.
Can you try keeping the current location of `makeNonNull(result, receiver);` and, before returning from this method, remove the `receiver` information from `result`, if the receiver is something other than a local variable (and/or other things?)?

Reply:
Sure

Reply:
What if the receiver is supposed to be non-null after calling a pure method, erasing the information from `result` would cause a false positive I think.

I think if the receiver is a map, and the method is `isMapGet`, we also don't want to remove the information.

Reply:
Yes, this only needs to happen for methods that are not pure or side-effect-free, good point!